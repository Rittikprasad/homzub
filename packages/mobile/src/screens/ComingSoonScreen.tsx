import React from 'react';
import { View, StyleSheet } from 'react-native';
import { WithTranslation, withTranslation } from 'react-i18next';
import { CommonParamList } from '@homzhub/mobile/src/navigation/Common';
import { theme } from '@homzhub/common/src/styles/theme';
import { Text } from '@homzhub/common/src/components/atoms/Text';
import { UserScreen } from '@homzhub/mobile/src/components/HOC/UserScreen';
import { LocaleConstants } from '@homzhub/common/src/services/Localization/constants';
import { NavigationScreenProps, ScreensKeys } from '@homzhub/mobile/src/navigation/interfaces';

type libraryProps = NavigationScreenProps<CommonParamList, ScreensKeys.ComingSoonScreen>;

type Props = WithTranslation & libraryProps;

class ComingSoonScreen extends React.PureComponent<Props> {
  public render = (): React.ReactElement => {
    const {
      t,
      route: {
        params: { title, tabHeader, message },
      },
    } = this.props;

    return (
      <UserScreen title={tabHeader} pageTitle={title} onBackPress={this.goBack}>
        <View style={[styles.screen, !!message && styles.extraStyle]}>
          <Text type={message ? 'small' : 'large'} textType="semiBold">
            {message ?? t('comingSoonText')}
          </Text>
        </View>
      </UserScreen>
    );
  };

  private goBack = (): void => {
    const { navigation } = this.props;
    navigation.goBack();
  };
}

export default withTranslation(LocaleConstants.namespacesKey.common)(ComingSoonScreen);

const styles = StyleSheet.create({
  screen: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: theme.colors.white,
    paddingTop: theme.viewport.height / 3,
    paddingBottom: theme.viewport.height / 3,
  },
  extraStyle: {
    paddingTop: theme.viewport.height / 4,
  },
});
