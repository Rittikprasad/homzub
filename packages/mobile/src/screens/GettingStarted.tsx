import React from 'react';
import { StyleSheet, View } from 'react-native';
import { withTranslation, WithTranslation } from 'react-i18next';
import { theme } from '@homzhub/common/src/styles/theme';
import { Button } from '@homzhub/common/src/components/atoms/Button';
import { Image } from '@homzhub/common/src/components/atoms/Image';
import { Text, Label } from '@homzhub/common/src/components/atoms/Text';
import { GuestStackNavigatorParamList } from '@homzhub/mobile/src/navigation/GuestStack';
import { NavigationScreenProps, ScreensKeys } from '@homzhub/mobile/src/navigation/interfaces';

type IProps = NavigationScreenProps<GuestStackNavigatorParamList, ScreensKeys.GettingStarted> & WithTranslation;

export class GettingStarted extends React.PureComponent<IProps> {
  public render(): React.ReactElement {
    const { t } = this.props;
    return (
      <>
        <View style={styles.container}>
          <Image source={{ uri: '' }} style={styles.image} />
          <Text type="regular" textType="semiBold" style={styles.header}>
            {t('header')}
          </Text>
          <View style={styles.buttonContainer}>
            <Button
              title={t('common:searchProperties')}
              type="primary"
              textSize="small"
              containerStyle={styles.searchProperty}
              onPress={this.searchProperty}
              testID="btnSearchProperty"
            />
            <Button
              title={t('login')}
              type="secondary"
              containerStyle={styles.login}
              onPress={this.login}
              testID="btnLogin"
            />
            <Label type="large" textType="regular" style={styles.image}>
              {t('newAroundHere')} &nbsp;
              <Label
                type="large"
                textType="bold"
                style={styles.signUpLink}
                onPress={this.navigateToSignUp}
                testID="lblSignup"
              >
                {t('signUp')}
              </Label>
            </Label>
          </View>
        </View>
      </>
    );
  }

  public searchProperty = (): void => {
    const { navigation } = this.props;
    navigation.navigate(ScreensKeys.SearchStack);
  };

  public login = (): void => {
    const { navigation } = this.props;
    navigation.navigate(ScreensKeys.AuthStack, { screen: ScreensKeys.Login, params: {} });
  };

  public navigateToSignUp = (): void => {
    const { navigation } = this.props;
    navigation.navigate(ScreensKeys.AuthStack, { screen: ScreensKeys.SignUp, params: {} });
  };
}

export default withTranslation()(GettingStarted);
const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: theme.colors.background,
    justifyContent: 'space-around',
    padding: theme.layout.screenPadding,
  },
  image: {
    alignSelf: 'center',
  },
  header: {
    textAlign: 'center',
    lineHeight: 30,
  },
  buttonContainer: {
    alignItems: 'stretch',
    marginHorizontal: 20,
  },
  searchProperty: {
    flex: 0,
  },
  login: {
    flex: 0,
    marginVertical: 20,
  },
  signUpLink: {
    color: theme.colors.blue,
  },
});
