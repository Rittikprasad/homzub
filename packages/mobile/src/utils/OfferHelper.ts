import { FunctionUtils } from '@homzhub/common/src/utils/FunctionUtils';
import { NavigationService } from '@homzhub/mobile/src/services/NavigationService';
import { StoreProviderService } from '@homzhub/common/src/services/StoreProviderService';
import { OfferActions } from '@homzhub/common/src/modules/offers/actions';
import { OfferAction } from '@homzhub/common/src/domain/models/Offer';
import { ScreensKeys } from '@homzhub/mobile/src/navigation/interfaces';
import { OffersVisitsType } from '@homzhub/common/src/constants/Offers';
import { ListingType } from '@homzhub/common/src/domain/repositories/interfaces';

interface IVisitAction {
  type: OffersVisitsType;
  name: string;
  id: number;
  leaseListingId: number | null;
  saleListingId: number | null;
}

class OfferHelper {
  public handleOfferActions = (action: OfferAction): void => {
    const {
      navigation: { navigate },
    } = NavigationService;
    switch (action) {
      case OfferAction.ACCEPT:
        navigate(ScreensKeys.AcceptOffer);
        break;
      case OfferAction.REJECT:
        navigate(ScreensKeys.RejectOffer);
        break;
      case OfferAction.COUNTER:
        navigate(ScreensKeys.SubmitOffer);
        break;
      case OfferAction.CANCEL:
        navigate(ScreensKeys.CancelOffer);
        break;
      default:
        FunctionUtils.noop();
    }
  };

  public handleOfferVisitAction = (props: IVisitAction): void => {
    const store = StoreProviderService.getStore();
    const {
      navigation: { navigate },
    } = NavigationService;

    const { type, name, id, leaseListingId, saleListingId } = props;

    store.dispatch(
      OfferActions.setCurrentOfferPayload({
        type: leaseListingId ? ListingType.LEASE_LISTING : ListingType.SALE_LISTING,
        listingId: leaseListingId || saleListingId || 0,
      })
    );

    const param = {
      isFromPortfolio: true,
      isFromTenancies: false,
      screenTitle: name,
      propertyId: id,
    };

    switch (type) {
      case OffersVisitsType.offers:
        navigate(ScreensKeys.OfferDetail, param);
        break;
      case OffersVisitsType.visits:
        navigate(ScreensKeys.PropertyVisits, param);
        break;
      default:
        FunctionUtils.noop();
    }
  };
}

const offerHelperUtils = new OfferHelper();
export { offerHelperUtils as OfferHelper };
