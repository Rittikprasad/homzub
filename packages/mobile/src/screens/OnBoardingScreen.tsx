import React from 'react';
import { SafeAreaView, StyleSheet, View } from 'react-native';
import { connect } from 'react-redux';
import { bindActionCreators, Dispatch } from 'redux';
import { withTranslation, WithTranslation } from 'react-i18next';
import { AlertHelper } from '@homzhub/common/src/utils/AlertHelper';
import { ErrorUtils } from '@homzhub/common/src/utils/ErrorUtils';
import { PlatformUtils } from '@homzhub/common/src/utils/PlatformUtils';
import { UserActions } from '@homzhub/common/src/modules/user/actions';
import { CommonRepository } from '@homzhub/common/src/domain/repositories/CommonRepository';
import { NavigationScreenProps, ScreensKeys } from '@homzhub/mobile/src/navigation/interfaces';
import { GuestStackNavigatorParamList } from '@homzhub/mobile/src/navigation/GuestStack';
import { icons } from '@homzhub/common/src/assets/icon';
import { theme } from '@homzhub/common/src/styles/theme';
import { Button } from '@homzhub/common/src/components/atoms/Button';
import { StatusBar } from '@homzhub/mobile/src/components/atoms/StatusBar';
import { SVGUri } from '@homzhub/common/src/components/atoms/Svg';
import { Text } from '@homzhub/common/src/components/atoms/Text';
import { PaginationComponent, SnapCarousel } from '@homzhub/mobile/src/components';
import { BottomSheet } from '@homzhub/common/src/components/molecules/BottomSheet';
import { OnBoarding } from '@homzhub/common/src/domain/models/OnBoarding';
import { User, UserType } from '@homzhub/common/src/constants/OnBoarding';

interface IDispatchProps {
  updateOnBoarding: (isOnBoardingCompleted: boolean) => void;
  setAddPropertyFlow: (payload: boolean) => void;
}

interface IOnBoardingScreenState {
  activeSlide: number;
  data: OnBoarding[];
  isSheetVisible: boolean;
  selectedType: string;
  message: string;
}

type libraryProps = WithTranslation & NavigationScreenProps<GuestStackNavigatorParamList, ScreensKeys.OnBoarding>;
type Props = IDispatchProps & libraryProps;

export class OnBoardingScreen extends React.PureComponent<Props, IOnBoardingScreenState> {
  public state = {
    activeSlide: 0,
    data: [],
    isSheetVisible: false,
    selectedType: '',
    message: '',
  };

  public componentDidMount = async (): Promise<void> => {
    await this.getOnBoardingData();
  };

  public render(): React.ReactNode {
    const { t } = this.props;
    const { data, activeSlide, isSheetVisible, selectedType, message } = this.state;

    if (data.length === 0) {
      return null;
    }

    const currentSlide: OnBoarding = data[activeSlide];
    return (
      <>
        <StatusBar barStyle="dark-content" statusBarBackground={theme.colors.white} />
        <SafeAreaView style={styles.container}>
          <View style={styles.flexOne}>
            <Text type="small" textType="bold" style={styles.skip} onPress={this.handleSkip}>
              {t('login')}
            </Text>
            <View style={styles.textContainer}>
              <Text type="large" textType="bold" style={styles.title}>
                {currentSlide.title}
              </Text>
              <Text type="small" textType="regular" style={styles.description}>
                {currentSlide.description}
              </Text>
            </View>
            <SnapCarousel
              carouselData={data}
              carouselItem={this.renderCarouselItem}
              activeIndex={activeSlide}
              onSnapToItem={this.onSnapToItem}
              itemWidth={theme.viewport.width - 30}
              contentStyle={styles.imageView}
            />
            <PaginationComponent
              dotsLength={data.length}
              activeSlide={activeSlide}
              activeDotStyle={styles.activeDot}
              inactiveDotStyle={styles.inactiveDot}
            />
            <Text type="regular" textType="semiBold" style={styles.heading}>
              {t('wantTo')}
            </Text>
            <Button
              type="primary"
              title={t('property:addProperty')}
              icon={icons.plus}
              iconSize={26}
              iconColor={theme.colors.white}
              onPress={this.onAddProperty}
              containerStyle={styles.button}
              titleStyle={styles.buttonTitle}
              testID="btnAddProperty"
            />
            <Button
              type="secondary"
              title={t('property:searchProperty')}
              icon={icons.search}
              iconSize={20}
              iconColor={theme.colors.blue}
              titleStyle={styles.buttonTitle}
              onPress={this.searchProperty}
              containerStyle={styles.button}
              testID="btnSearchProperty"
            />
          </View>
          <BottomSheet visible={isSheetVisible} onCloseSheet={this.onCloseSheet} sheetHeight={500}>
            <>
              <Text type="regular" textType="semiBold" style={styles.sheetTitle}>
                {t('iAm')}
              </Text>
              <View style={styles.sheetContent}>
                {UserType.map((item) => {
                  const isSelected = selectedType === item.key;
                  return (
                    <Button
                      key={item.id}
                      type={isSelected ? 'primary' : 'secondary'}
                      fontType="regular"
                      title={t(item.name)}
                      icon={item.icon}
                      iconSize={24}
                      iconColor={isSelected ? theme.colors.white : theme.colors.darkTint3}
                      titleStyle={[
                        styles.selectionButton,
                        { color: isSelected ? theme.colors.white : theme.colors.darkTint2 },
                      ]}
                      onPress={(): void => this.onSelectType(item.key)}
                      containerStyle={styles.buttonContainer}
                      testID="btnSelect"
                    />
                  );
                })}
              </View>
              {!!message && (
                <Text type="small" textType="regular" style={styles.message}>
                  {message}
                </Text>
              )}
              <Button
                type="primary"
                title={t('continue')}
                disabled={!selectedType || selectedType !== User.OWNER}
                onPress={this.onContinue}
                containerStyle={styles.continueBtn}
                testID="btnContinue"
              />
            </>
          </BottomSheet>
        </SafeAreaView>
      </>
    );
  }

  // TODO: Refactor
  private renderCarouselItem = (item: OnBoarding): React.ReactElement => {
    const {
      viewport: { width, height: viewportHeight },
      DeviceDimensions: { SMALL, MEDIUM },
    } = theme;
    /* eslint-disable */
    const height =
      width > SMALL.width
        ? width > MEDIUM.width
          ? PlatformUtils.isIOS()
            ? 450
            : viewportHeight / 2.5
          : width === MEDIUM.width
          ? 280
          : 220
        : 150;
    /* eslint-enable */
    const imgWidth = width > SMALL.width ? (width > MEDIUM.width ? 320 : width === MEDIUM.width ? 350 : 330) : 280;
    return (
      <View style={styles.imageView}>
        <SVGUri uri={item.imageUrl} height={height} width={imgWidth} />
      </View>
    );
  };

  private onSelectType = (key: User): void => {
    const { t } = this.props;
    let message = '';
    switch (key) {
      case User.OWNER:
        message = t('auth:clickToSignUp');
        break;
      case User.AGENT:
      case User.MANAGER:
        message = t('auth:appNotReady');
        break;
      default:
        message = '';
    }
    this.setState({
      selectedType: key,
      message,
    });
  };

  private onContinue = (): void => {
    const { selectedType } = this.state;
    const { navigation, setAddPropertyFlow } = this.props;
    if (selectedType === User.OWNER) {
      navigation.navigate(ScreensKeys.AuthStack, { screen: ScreensKeys.SignUp, params: {} });
      setAddPropertyFlow(true);
      this.onCloseSheet();
    }
  };

  private onAddProperty = (): void => {
    const { navigation, setAddPropertyFlow } = this.props;
    navigation.navigate(ScreensKeys.AuthStack, { screen: ScreensKeys.SignUp, params: {} });
    setAddPropertyFlow(true);
  };

  private onCloseSheet = (): void => {
    this.setState({
      isSheetVisible: false,
    });
  };

  private onSnapToItem = (slideNumber: number): void => {
    this.setState({ activeSlide: slideNumber });
  };

  private handleSkip = (): void => {
    const { navigation } = this.props;
    navigation.navigate(ScreensKeys.AuthStack, { screen: ScreensKeys.Login, params: {} });
  };

  private getOnBoardingData = async (): Promise<void> => {
    try {
      const response = await CommonRepository.getOnBoarding();
      this.setState({
        data: response,
      });
    } catch (error) {
      AlertHelper.error({ message: ErrorUtils.getErrorMessage(error.details) });
    }
  };

  private searchProperty = (): void => {
    const { navigation } = this.props;
    navigation.navigate(ScreensKeys.SearchStack);
  };
}

const styles = StyleSheet.create({
  flexOne: {
    flex: 1,
  },
  container: {
    flex: 1,
    backgroundColor: theme.colors.white,
  },
  textContainer: {
    paddingTop: 35,
    alignItems: 'center',
  },
  description: {
    textAlign: 'center',
    marginHorizontal: 10,
  },
  button: {
    flex: 0,
    marginHorizontal: 36,
    marginBottom: 10,
    flexDirection: 'row-reverse',
  },
  buttonContainer: {
    width: 90,
    flex: 0,
    marginRight: 10,
    marginBottom: 10,
    paddingVertical: 15,
    borderColor: theme.colors.disabled,
    flexDirection: 'column-reverse',
  },
  activeDot: {
    width: 16,
    backgroundColor: theme.colors.blue,
    borderColor: theme.colors.blue,
  },
  inactiveDot: {
    backgroundColor: theme.colors.darkTint9,
    borderColor: theme.colors.darkTint9,
    borderRadius: 5,
    width: 10,
    height: 10,
  },
  title: {
    color: theme.colors.blackTint1,
    padding: 10,
    fontSize: 24,
    textAlign: 'center',
    lineHeight: 28,
  },
  heading: {
    color: theme.colors.darkTint3,
    alignSelf: 'center',
    paddingBottom: 16,
  },
  buttonTitle: {
    marginHorizontal: 12,
  },
  sheetTitle: {
    textAlign: 'center',
    marginBottom: 20,
  },
  sheetContent: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    marginBottom: 50,
    marginHorizontal: 20,
  },
  selectionButton: {
    marginHorizontal: 0,
    width: 100,
  },
  message: {
    textAlign: 'center',
    color: theme.colors.darkTint5,
    marginBottom: 8,
    paddingHorizontal: 10,
  },
  continueBtn: {
    flex: 0,
    justifyContent: 'center',
    marginHorizontal: 30,
    height: 50,
  },
  skip: {
    color: theme.colors.blue,
    alignSelf: 'flex-end',
    marginHorizontal: 20,
  },
  imageView: {
    alignItems: 'center',
    marginTop: 10,
  },
});

export const mapDispatchToProps = (dispatch: Dispatch): IDispatchProps => {
  const { updateOnBoarding, setAddPropertyFlow } = UserActions;
  return bindActionCreators(
    {
      updateOnBoarding,
      setAddPropertyFlow,
    },
    dispatch
  );
};

export default connect(null, mapDispatchToProps)(withTranslation()(OnBoardingScreen));
