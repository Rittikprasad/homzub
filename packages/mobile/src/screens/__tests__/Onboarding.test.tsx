import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { ObjectMapper } from '@homzhub/common/src/utils/ObjectMapper';
import { CommonRepository } from '@homzhub/common/src/domain/repositories/CommonRepository';
import { UserActionTypes } from '@homzhub/common/src/modules/user/actions';
import { OnboardingData } from '@homzhub/common/src/mocks/Onboarding';
import { OnBoardingScreen, mapDispatchToProps } from '@homzhub/mobile/src/screens/OnBoardingScreen';
import { OnBoarding } from '@homzhub/common/src/domain/models/OnBoarding';

const mock = jest.fn();

describe('Onboarding Screen', () => {
  let component: any;
  let props: any;
  let instance: any;

  beforeEach(() => {
    props = {
      navigation: { navigate: mock },
      updateOnBoarding: mock,
    };
    component = shallow(<OnBoardingScreen {...props} t={(key: string): string => key} />);
    instance = component.instance();
  });

  afterAll(() => {
    jest.clearAllMocks();
  });

  it('should match snapshot', () => {
    expect(toJson(component)).toMatchSnapshot();
  });

  it('should render with data in state', () => {
    instance.setState({ data: OnboardingData, activeSlide: 0 });
    expect(toJson(component)).toMatchSnapshot();
  });

  it('should render without data in state', () => {
    expect(toJson(component)).toMatchSnapshot();
  });

  it('should render got it button for active slide', () => {
    instance.setState({ data: OnboardingData, activeSlide: 2 });
    expect(toJson(component)).toMatchSnapshot();
  });

  it('should call the update on boarding', () => {
    const dispatch = jest.fn();
    mapDispatchToProps(dispatch).updateOnBoarding(true);
    expect(dispatch.mock.calls[0][0]).toEqual({
      type: UserActionTypes.UPDATE.ONBOARDING,
      payload: true,
    });
  });

  it('should call get onboarding data', async () => {
    const deserializedData = ObjectMapper.deserializeArray(OnBoarding, OnboardingData);
    jest.spyOn(CommonRepository, 'getOnBoarding').mockImplementation(async () => Promise.resolve(deserializedData));
    await instance.componentDidMount();
    const response = await CommonRepository.getOnBoarding();
    instance.setState({ data: response });
    expect(component.state('data')).toBe(deserializedData);
  });
});
