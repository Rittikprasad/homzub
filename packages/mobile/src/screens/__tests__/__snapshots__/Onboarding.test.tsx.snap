// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Onboarding Screen should match snapshot 1`] = `""`;

exports[`Onboarding Screen should render got it button for active slide 1`] = `
<Fragment>
  <Memo(HomzhubStatusBar)
    barStyle="dark-content"
    statusBarBackground="#FFFFFF"
  />
  <ForwardRef(SafeAreaView)
    style={
      Object {
        "backgroundColor": "#FFFFFF",
        "flex": 1,
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <Text
        onPress={[Function]}
        style={
          Object {
            "alignSelf": "flex-end",
            "color": "#0084F8",
            "marginHorizontal": 20,
          }
        }
        textType="bold"
        type="small"
      >
        login
      </Text>
      <View
        style={
          Object {
            "alignItems": "center",
            "paddingTop": 35,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#263238",
              "fontSize": 24,
              "lineHeight": 28,
              "padding": 10,
              "textAlign": "center",
            }
          }
          textType="bold"
          type="large"
        >
          Maintain
        </Text>
        <Text
          style={
            Object {
              "marginHorizontal": 10,
              "textAlign": "center",
            }
          }
          textType="regular"
          type="small"
        >
          Sit amet, in enim nunc, etiam amet velit. Malesuada non neque nec in a.
        </Text>
      </View>
      <SnapCarousel
        activeIndex={2}
        carouselData={
          Array [
            Object {
              "description": "Sit amet, in enim nunc, etiam amet velit. Malesuada non neque nec in a.",
              "image_url": "https://homzhub-bucket.s3.amazonaws.com/Search_Onboarding.svg",
              "title": "Search",
            },
            Object {
              "description": "Sit amet, in enim nunc, etiam amet velit. Malesuada non neque nec in a.",
              "image_url": "https://homzhub-bucket.s3.amazonaws.com/Post_Onboarding.svg",
              "title": "Post",
            },
            Object {
              "description": "Sit amet, in enim nunc, etiam amet velit. Malesuada non neque nec in a.",
              "image_url": "https://homzhub-bucket.s3.amazonaws.com/Maintain_Onboarding.svg",
              "title": "Maintain",
            },
          ]
        }
        carouselItem={[Function]}
        contentStyle={
          Object {
            "alignItems": "center",
            "marginTop": 10,
          }
        }
        itemWidth={720}
        onSnapToItem={[Function]}
      />
      <PaginationComponent
        activeDotStyle={
          Object {
            "backgroundColor": "#0084F8",
            "borderColor": "#0084F8",
            "width": 16,
          }
        }
        activeSlide={2}
        dotsLength={3}
        inactiveDotStyle={
          Object {
            "backgroundColor": "#CCD1D7",
            "borderColor": "#CCD1D7",
            "borderRadius": 5,
            "height": 10,
            "width": 10,
          }
        }
      />
      <Text
        style={
          Object {
            "alignSelf": "center",
            "color": "#33495E",
            "paddingBottom": 16,
          }
        }
        textType="semiBold"
        type="regular"
      >
        wantTo
      </Text>
      <Button
        containerStyle={
          Object {
            "flex": 0,
            "flexDirection": "row-reverse",
            "marginBottom": 10,
            "marginHorizontal": 36,
          }
        }
        icon="plus"
        iconColor="#FFFFFF"
        iconSize={26}
        onPress={[Function]}
        testID="btnAddProperty"
        title="property:addProperty"
        titleStyle={
          Object {
            "marginHorizontal": 12,
          }
        }
        type="primary"
      />
      <Button
        containerStyle={
          Object {
            "flex": 0,
            "flexDirection": "row-reverse",
            "marginBottom": 10,
            "marginHorizontal": 36,
          }
        }
        icon="search"
        iconColor="#0084F8"
        iconSize={20}
        onPress={[Function]}
        testID="btnSearchProperty"
        title="property:searchProperty"
        titleStyle={
          Object {
            "marginHorizontal": 12,
          }
        }
        type="secondary"
      />
    </View>
    <BottomSheet
      onCloseSheet={[Function]}
      sheetHeight={500}
      visible={false}
    >
      <Text
        style={
          Object {
            "marginBottom": 20,
            "textAlign": "center",
          }
        }
        textType="semiBold"
        type="regular"
      >
        iAm
      </Text>
      <View
        style={
          Object {
            "flexDirection": "row",
            "justifyContent": "space-between",
            "marginBottom": 50,
            "marginHorizontal": 20,
          }
        }
      >
        <Button
          containerStyle={
            Object {
              "borderColor": "#C9D6DF",
              "flex": 0,
              "flexDirection": "column-reverse",
              "marginBottom": 10,
              "marginRight": 10,
              "paddingVertical": 15,
              "width": 90,
            }
          }
          fontType="regular"
          icon="owner"
          iconColor="#33495E"
          iconSize={24}
          key="1"
          onPress={[Function]}
          testID="btnSelect"
          title="property:propertyOwner"
          titleStyle={
            Array [
              Object {
                "marginHorizontal": 0,
                "width": 100,
              },
              Object {
                "color": "#19324A",
              },
            ]
          }
          type="secondary"
        />
        <Button
          containerStyle={
            Object {
              "borderColor": "#C9D6DF",
              "flex": 0,
              "flexDirection": "column-reverse",
              "marginBottom": 10,
              "marginRight": 10,
              "paddingVertical": 15,
              "width": 90,
            }
          }
          fontType="regular"
          icon="agent"
          iconColor="#33495E"
          iconSize={24}
          key="2"
          onPress={[Function]}
          testID="btnSelect"
          title="property:propertyAgent"
          titleStyle={
            Array [
              Object {
                "marginHorizontal": 0,
                "width": 100,
              },
              Object {
                "color": "#19324A",
              },
            ]
          }
          type="secondary"
        />
        <Button
          containerStyle={
            Object {
              "borderColor": "#C9D6DF",
              "flex": 0,
              "flexDirection": "column-reverse",
              "marginBottom": 10,
              "marginRight": 10,
              "paddingVertical": 15,
              "width": 90,
            }
          }
          fontType="regular"
          icon="manager"
          iconColor="#33495E"
          iconSize={24}
          key="3"
          onPress={[Function]}
          testID="btnSelect"
          title="property:propertyManager"
          titleStyle={
            Array [
              Object {
                "marginHorizontal": 0,
                "width": 100,
              },
              Object {
                "color": "#19324A",
              },
            ]
          }
          type="secondary"
        />
      </View>
      <Button
        containerStyle={
          Object {
            "flex": 0,
            "height": 50,
            "justifyContent": "center",
            "marginHorizontal": 30,
          }
        }
        disabled={true}
        onPress={[Function]}
        testID="btnContinue"
        title="continue"
        type="primary"
      />
    </BottomSheet>
  </ForwardRef(SafeAreaView)>
</Fragment>
`;

exports[`Onboarding Screen should render with data in state 1`] = `
<Fragment>
  <Memo(HomzhubStatusBar)
    barStyle="dark-content"
    statusBarBackground="#FFFFFF"
  />
  <ForwardRef(SafeAreaView)
    style={
      Object {
        "backgroundColor": "#FFFFFF",
        "flex": 1,
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <Text
        onPress={[Function]}
        style={
          Object {
            "alignSelf": "flex-end",
            "color": "#0084F8",
            "marginHorizontal": 20,
          }
        }
        textType="bold"
        type="small"
      >
        login
      </Text>
      <View
        style={
          Object {
            "alignItems": "center",
            "paddingTop": 35,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#263238",
              "fontSize": 24,
              "lineHeight": 28,
              "padding": 10,
              "textAlign": "center",
            }
          }
          textType="bold"
          type="large"
        >
          Search
        </Text>
        <Text
          style={
            Object {
              "marginHorizontal": 10,
              "textAlign": "center",
            }
          }
          textType="regular"
          type="small"
        >
          Sit amet, in enim nunc, etiam amet velit. Malesuada non neque nec in a.
        </Text>
      </View>
      <SnapCarousel
        activeIndex={0}
        carouselData={
          Array [
            Object {
              "description": "Sit amet, in enim nunc, etiam amet velit. Malesuada non neque nec in a.",
              "image_url": "https://homzhub-bucket.s3.amazonaws.com/Search_Onboarding.svg",
              "title": "Search",
            },
            Object {
              "description": "Sit amet, in enim nunc, etiam amet velit. Malesuada non neque nec in a.",
              "image_url": "https://homzhub-bucket.s3.amazonaws.com/Post_Onboarding.svg",
              "title": "Post",
            },
            Object {
              "description": "Sit amet, in enim nunc, etiam amet velit. Malesuada non neque nec in a.",
              "image_url": "https://homzhub-bucket.s3.amazonaws.com/Maintain_Onboarding.svg",
              "title": "Maintain",
            },
          ]
        }
        carouselItem={[Function]}
        contentStyle={
          Object {
            "alignItems": "center",
            "marginTop": 10,
          }
        }
        itemWidth={720}
        onSnapToItem={[Function]}
      />
      <PaginationComponent
        activeDotStyle={
          Object {
            "backgroundColor": "#0084F8",
            "borderColor": "#0084F8",
            "width": 16,
          }
        }
        activeSlide={0}
        dotsLength={3}
        inactiveDotStyle={
          Object {
            "backgroundColor": "#CCD1D7",
            "borderColor": "#CCD1D7",
            "borderRadius": 5,
            "height": 10,
            "width": 10,
          }
        }
      />
      <Text
        style={
          Object {
            "alignSelf": "center",
            "color": "#33495E",
            "paddingBottom": 16,
          }
        }
        textType="semiBold"
        type="regular"
      >
        wantTo
      </Text>
      <Button
        containerStyle={
          Object {
            "flex": 0,
            "flexDirection": "row-reverse",
            "marginBottom": 10,
            "marginHorizontal": 36,
          }
        }
        icon="plus"
        iconColor="#FFFFFF"
        iconSize={26}
        onPress={[Function]}
        testID="btnAddProperty"
        title="property:addProperty"
        titleStyle={
          Object {
            "marginHorizontal": 12,
          }
        }
        type="primary"
      />
      <Button
        containerStyle={
          Object {
            "flex": 0,
            "flexDirection": "row-reverse",
            "marginBottom": 10,
            "marginHorizontal": 36,
          }
        }
        icon="search"
        iconColor="#0084F8"
        iconSize={20}
        onPress={[Function]}
        testID="btnSearchProperty"
        title="property:searchProperty"
        titleStyle={
          Object {
            "marginHorizontal": 12,
          }
        }
        type="secondary"
      />
    </View>
    <BottomSheet
      onCloseSheet={[Function]}
      sheetHeight={500}
      visible={false}
    >
      <Text
        style={
          Object {
            "marginBottom": 20,
            "textAlign": "center",
          }
        }
        textType="semiBold"
        type="regular"
      >
        iAm
      </Text>
      <View
        style={
          Object {
            "flexDirection": "row",
            "justifyContent": "space-between",
            "marginBottom": 50,
            "marginHorizontal": 20,
          }
        }
      >
        <Button
          containerStyle={
            Object {
              "borderColor": "#C9D6DF",
              "flex": 0,
              "flexDirection": "column-reverse",
              "marginBottom": 10,
              "marginRight": 10,
              "paddingVertical": 15,
              "width": 90,
            }
          }
          fontType="regular"
          icon="owner"
          iconColor="#33495E"
          iconSize={24}
          key="1"
          onPress={[Function]}
          testID="btnSelect"
          title="property:propertyOwner"
          titleStyle={
            Array [
              Object {
                "marginHorizontal": 0,
                "width": 100,
              },
              Object {
                "color": "#19324A",
              },
            ]
          }
          type="secondary"
        />
        <Button
          containerStyle={
            Object {
              "borderColor": "#C9D6DF",
              "flex": 0,
              "flexDirection": "column-reverse",
              "marginBottom": 10,
              "marginRight": 10,
              "paddingVertical": 15,
              "width": 90,
            }
          }
          fontType="regular"
          icon="agent"
          iconColor="#33495E"
          iconSize={24}
          key="2"
          onPress={[Function]}
          testID="btnSelect"
          title="property:propertyAgent"
          titleStyle={
            Array [
              Object {
                "marginHorizontal": 0,
                "width": 100,
              },
              Object {
                "color": "#19324A",
              },
            ]
          }
          type="secondary"
        />
        <Button
          containerStyle={
            Object {
              "borderColor": "#C9D6DF",
              "flex": 0,
              "flexDirection": "column-reverse",
              "marginBottom": 10,
              "marginRight": 10,
              "paddingVertical": 15,
              "width": 90,
            }
          }
          fontType="regular"
          icon="manager"
          iconColor="#33495E"
          iconSize={24}
          key="3"
          onPress={[Function]}
          testID="btnSelect"
          title="property:propertyManager"
          titleStyle={
            Array [
              Object {
                "marginHorizontal": 0,
                "width": 100,
              },
              Object {
                "color": "#19324A",
              },
            ]
          }
          type="secondary"
        />
      </View>
      <Button
        containerStyle={
          Object {
            "flex": 0,
            "height": 50,
            "justifyContent": "center",
            "marginHorizontal": 30,
          }
        }
        disabled={true}
        onPress={[Function]}
        testID="btnContinue"
        title="continue"
        type="primary"
      />
    </BottomSheet>
  </ForwardRef(SafeAreaView)>
</Fragment>
`;

exports[`Onboarding Screen should render without data in state 1`] = `""`;
